client-cli/src/main/java/org/eclipse/dataspaceconnector/registration/cli/RegistrationUrlResolver.java

```diff
@@ -0,0 +1,53 @@
+/*
+ *  Copyright (c) 2022 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.registration.cli;
+
+import org.eclipse.dataspaceconnector.iam.did.spi.document.DidDocument;
+import org.eclipse.dataspaceconnector.iam.did.spi.document.Service;
+import org.eclipse.dataspaceconnector.iam.did.spi.resolution.DidResolver;
+import org.eclipse.dataspaceconnector.spi.result.Result;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Optional;
+
+/**
+ * Resolves the registration url from the DID.
+ */
+public class RegistrationUrlResolver {
+
+    public static final String REGISTRATION_URL = "RegistrationUrl";
+
+    /**
+     * Resolves the DID document from did:web.
+     */
+    private final DidResolver resolver;
+
+    public RegistrationUrlResolver(DidResolver resolver) {
+        this.resolver = resolver;
+    }
+
+    @NotNull
+    public Optional<String> resolveUrl(String did) {
```

Comment:
Not a big fan of using `Optional` as method return value. I'd prefer returning a `@Nullable` String or, even better, a `Result<String>`